### PR TITLE
xplat: xcode 8 on capitan std::string references + enable verbose for MAKE

### PR DESCRIPTION
**Pull request by obastemur**
- xcode 8.0 on capitan asks for undefined std::string::compare reference on ch.

- Enabling verbose for MAKE too.

- fix `--no-jit` build

### CODE REVIEW

**Comment by obastemur** [comment]
@dotnet-bot test ci please

**Comment by obastemur** [comment]
@dotnet-bot test ci please

**Comment by obastemur** [comment]
@dotnet-bot test ci please

**Comment by obastemur** [comment]
This PR also adds `--test-build` + `shared` + `--no-jit` test to CI (ubuntu + osx).

We weren't testing OSX shared builds and `--no-jit` option.

/cc @dilijev

File: bin/ch/CMakeLists.txt [hunk 1]
```diff
@@ -69,14 +69,20 @@ if(STATIC_LIBRARY)
...
   set(lib_target
     PRIVATE Chakra.Pal
     PRIVATE Chakra.Common.Codex.Singular
     PRIVATE Chakra.Runtime.PlatformAgnostic.Singular
     )
+  if(CMAKE_SYSTEM_NAME STREQUAL Darwin)
+    set(lib_target "${lib_target}"
+      "-stdlib=libstdc++" # expected 'deprecated' warning. needed for std::string
+      )
+  endif()
```
Comment (jianchun) [inline bin/ch/CMakeLists.txt]: Why not move this out of "if staticlib else" to top level, so that you don't need to duplicate?

**Comment by jianchun** [comment]
> VERBOSE=""

Looks this should have been "_VERBOSE"?

---
Refers to: build.sh:63 in 31a439e. [](commit_id = 31a439ef21557e1f2e411e0658c06f7a0f6a6357, deletion_comment = False)

File: bin/ch/CMakeLists.txt (hunk 1 above)
Comment (obastemur) [inline bin/ch/CMakeLists.txt]: order ?

File: netci.groovy [hunk 2]
```diff
@@ -114,66 +114,77 @@ def CreateBuildTasks = { machine, configTag, buildExtra, testExtra, runCodeAnaly
...
+
+    config = (configTag == null) ? config : "${configTag}_${config}"
+    config = staticBuild ? "${config}_static" : config
+
+    // params: Project, BaseTaskName, IsPullRequest (appends '_prtest')
+    def jobName = Utilities.getFullJobName(project, config, isPR) + customOption.replaceAll(/[-]+/, "_")
+
+    def infoScript = "bash jenkins/get_system_info.sh --${platform}"
+    def buildFlag = buildType == "release" ? "" : (buildType == "debug" ? "--debug" : "--test-build")
+    staticBuild = buildType == "debug" && platform == "osx" ? "" : staticBuild
```
Comment (jianchun) [inline netci.groovy]: Changing "staticBuild" after composing "jobName". "jobName" will be wrong.

Comment (obastemur) [inline netci.groovy]: that line is not needed though. removing it

**Comment by obastemur** [comment]
@dotnet-bot test ci please

**Comment by jianchun** [comment]
:shipit:

**Comment by obastemur** [comment]
@jianchun Thanks for the review

**Pull request merged**